src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -180,779 +180,1127 @@ void window_scenery_update_scroll(rct_window* w);
+                        }
+                            
+                        windowHeight = std::max(WINDOW_SCENERY_HEIGHT, windowHeight);
 
-    for (int32_t scenerySetId = 0; scenerySetId < MAX_SCENERY_GROUP_OBJECTS; scenerySetId++)
+                        min_width = WINDOW_SCENERY_WIDTH;
+                        max_width = WINDOW_SCENERY_WIDTH;
+                        min_height = windowHeight;
+                        max_height = windowHeight;
+                    }
+                }
+            }
+        }
+        else
+        {
+            scenery.hover_counter = 0;
+
+            if (input_get_state() != InputState::ScrollLeft)
+            {
+                min_width = WINDOW_SCENERY_WIDTH;
+                max_width = WINDOW_SCENERY_WIDTH;
+                min_height = WINDOW_SCENERY_HEIGHT;
+                max_height = WINDOW_SCENERY_HEIGHT;
+            }
+        }
+
+        Invalidate();
+
+        if (!scenery_tool_is_active())
+        {
+            Close();
+            return;
+        }
+
+        if (gWindowSceneryEyedropperEnabled)
+        {
+            gCurrentToolId = Tool::Crosshair;
+        }
+        else if (gWindowSceneryPaintEnabled == 1)
+        {
+            gCurrentToolId = Tool::PaintDown;
+        }
+        else
+        {
+            uint16_t tabIndex = gWindowSceneryActiveTabIndex;
+            ScenerySelection tabSelectedScenery = gWindowSceneryTabSelections[tabIndex];
+
+            if (!tabSelectedScenery.IsUndefined())
+            {
+                if (tabSelectedScenery.SceneryType == SCENERY_TYPE_BANNER)
+                {
+                    gCurrentToolId = Tool::EntranceDown;
+                }
+                else if (tabSelectedScenery.SceneryType == SCENERY_TYPE_LARGE)
+                {
+                    gCurrentToolId = static_cast<Tool>(
+                        get_large_scenery_entry(tabSelectedScenery.EntryIndex)->large_scenery.tool_id);
+                }
+                else if (tabSelectedScenery.SceneryType == SCENERY_TYPE_WALL)
+                {
+                    gCurrentToolId = static_cast<Tool>(get_wall_entry(tabSelectedScenery.EntryIndex)->wall.tool_id);
+                }
+                else if (tabSelectedScenery.SceneryType == SCENERY_TYPE_PATH_ITEM)
+                { // path bit
+                    gCurrentToolId = static_cast<Tool>(
+                        get_footpath_item_entry(tabSelectedScenery.EntryIndex)->path_bit.tool_id);
+                }
+                else
+                { // small scenery
+                    gCurrentToolId = static_cast<Tool>(
+                        get_small_scenery_entry(tabSelectedScenery.EntryIndex)->small_scenery.tool_id);
+                }
+            }
+        }
+    }
+
+    /**
+     *
+     *  rct2: 0x006E1A91
+     */
+    ScreenSize OnScrollGetSize(int32_t scrollIndex /*, int32_t* width, int32_t* height */) override // TODO: width/height
```

Comment:
I've figured out that width and height are overridable with the new Window implementation. It seems like I only care about height here though and I can't have it ignore width by doing something like:

```
int32_t rows = window_scenery_count_rows();
auto height = window_scenery_rows_height(rows);

return { nullptr, height };
```

Any suggestions?

Reply:
yes, just return 0